voidcrew/code/modules/overmap/ships/pvp/subverter.dm

```diff
@@ -0,0 +1,329 @@
+#define SUB_RECHARGING 4
+#define SUB_TARGET_SUBVERTED 5
+#define SUB_TARGET_DOCKED 6
+#define SUB_TARGET_DOCKING 7
+#define SUB_TARGET_UNDOCKING 8
+#define SUB_TARGET_GRACE 9
+#define SUB_TARGET_ANTIVIRUS 10
+
+/obj/machinery/subverter
+	name = "interdictor"
+	desc = "A piece of equipment used to upload programs to other vessels. It has no interface: It could be linked to an auxiliary console or the wires could be pulsed manually."
+	icon = 'voidcrew/icons/obj/machines/space_hacking.dmi'
+	icon_state = "subverter"
+	density = TRUE
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 100
+	active_power_usage = 1000
+	circuit = /obj/item/circuitboard/machine/subverter
+	layer = BELOW_OBJ_LAYER
+	//Which ship the subverter is on
+	var/obj/structure/overmap/ship/simulated/ship
+	//Linked auxiliary console
+	var/obj/machinery/computer/autopilot/aux
+	//Standard machine stuff
+	var/hacked = FALSE
+	var/disabled = FALSE
+	var/shocked = FALSE
+	//Cooldown that handles the subverter recharging
+	COOLDOWN_DECLARE(subverter_cooldown)
+	//Used to handle higher tier parts affecting the subverter
+	var/sub_recharge = SUBVERTER_RECHARGE_TIME
+	var/sub_engine = SUBVERTER_ENGINE_STALL_TIME
+	var/spiciness = SUBVERTER_SPICINESS
+
+/obj/machinery/subverter/Initialize()
+	. = ..()
+	wires = new /datum/wires/subverter(src)
+	update_stats()
+	COOLDOWN_START(src, subverter_cooldown, sub_recharge)
+
+/obj/machinery/subverter/connect_to_shuttle(obj/docking_port/mobile/port, obj/docking_port/stationary/dock)
+	ship = port.current_ship
+
+/obj/machinery/subverter/on_deconstruction()
+	if (aux)
+		aux.sub = null
+		aux = null
+	..()
+
+/**
+*	Returns formatted remaining time till the subverter finishes recharging
+*/
+/obj/machinery/subverter/proc/get_recharge_str()
+	. = COOLDOWN_TIMELEFT(src, subverter_cooldown) / 10
+	. *= . > 0
+	return "[add_leading(num2text((. / 60) % 60), 2, "0")]:[add_leading(num2text(. % 60), 2, "0")]"
+
+/**
+*	Causes subverter to make a certain sound and the aux console to display a certain message
+*	Exist to prevent player confusion when the subversion attempt fails
+*
+*	message - what the aux console will say
+*	soundin - sound the subverter will make
+*/
+/obj/machinery/subverter/proc/subvert_bark(message, soundin = SUBVERTER_BARK_FAILURE)
+	if (aux)
+		aux.say(message)
+	playsound(loc, soundin, 90, 1, 0)
+
+/**
+*	This handles the different contexts a subverter could bark in response to.
+*	It will call subvert_bark with parameters that fit the situation.
+*
+*	subvert_flag - The context of the subversion attempt, returned by can_subvert
+*	obj/structure/overmap/ship/simulated/target_ship - The target ship of the subversion attempt
+*	bark_success_message - (optional) Text displayed upon a successful subversion
+*/
+/obj/machinery/subverter/proc/bark_processing(subvert_flag, obj/structure/overmap/ship/simulated/target_ship, bark_success_message = "Agent was successful! Bringing [target_ship.name] out of hyperspace.")
```

Comment:
you don't need `target_ship` anymore

Reply:
i do actually need that for the success message but im just gonna edit it so it just says "target vessel" like the other messages

Reply:
oh i didn't see it in the message whoops